### PR TITLE
fix: trader on BSC and Matic network

### PR DESCRIPTION
**Pull request by guanbinrui**
closes #

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

**Review by septs — DISMISSED** [review]

File: packages/maskbook/src/plugins/Trader/helpers/uniswap.ts [hunk 1]
```diff
@@ -67,6 +65,45 @@ export function uniswapTokenTo(token: UniswapToken) {
...
+    public get wrapped(): Token {
+        if (this.chainId in WETH) return toUniswapToken(this.chainId, WETH[this.chainId as ChainId])
+        throw new Error('Unsupported chain ID')
+    }
+
+    private static _cachedEther: { [chainId: number]: ExtendedEther } = {}
+
+    public static onChain(chainId: number): ExtendedEther {
+        return this._cachedEther[chainId] ?? (this._cachedEther[chainId] = new ExtendedEther(chainId))
+    }
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/helpers/uniswap.ts:108]:
```suggestion
    }
    return
```

File: .vscode/schema/constants.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "$schema": "http://json-schema.org/draft-07/schema",
     "type": "object",
     "propertyNames": {
-        "pattern": "[A-F_]+"
+        "pattern": "[A-Za-z_]+$"
```
Comment (septs) [inline .vscode/schema/constants.json:5]:
```suggestion
        "pattern": "^[A-Za-z_]+$"
```

**Review by septs — DISMISSED** [review]

**Review by septs — DISMISSED** [review]

**Comment by Jack-Works** [comment]
Linters and tests / type-check (pull_request) Failing after 2m — type-check

**Comment by guanbinrui** [comment]
> Linters and tests / type-check (pull_request) Failing after 2m — type-check

FIXED

**Review by septs — APPROVED** [review]

**Pull request merged**